contracts/liquidity_hub/fee_collector/src/queries.rs

```diff
@@ -12,3 +17,153 @@ pub fn query_config(deps: Deps) -> StdResult<ConfigResponse> {
     let config = CONFIG.load(deps.storage)?;
     Ok(config)
 }
+
+pub fn query_fees(
+    deps: Deps,
+    collect_fees_for: QueryFeesFor,
+    accrued: bool,
+) -> StdResult<Vec<Asset>> {
+    let mut query_fees_messages: Vec<Asset> = Vec::new();
+
+    match collect_fees_for {
+        QueryFeesFor::Contracts { contracts } => {
+            for contract in contracts {
+                match contract.contract_type {
+                    ContractType::Pool {} => {
+                        let mut pair_fee =
+                            query_fees_for_pair(&deps, contract.address.clone(), accrued)?;
+
+                        query_fees_messages.append(&mut pair_fee);
+                    }
+                    ContractType::Vault {} => {
+                        let vault_fee =
+                            query_fees_for_vault(&deps, contract.address.clone(), accrued)?;
+
+                        query_fees_messages.push(vault_fee);
+                    }
+                }
+            }
+        }
+        QueryFeesFor::Factory {
+            factory_addr,
+            factory_type,
+        } => {
+            let factory = deps.api.addr_validate(factory_addr.as_str())?;
+            let mut assets = query_fees_for_factory(&deps, &factory, factory_type, accrued)?;
+
+            query_fees_messages.append(&mut assets);
+        }
+    }
+
+    Ok(query_fees_messages)
+}
+
+fn query_fees_for_vault(deps: &Deps, vault: String, accrued: bool) -> StdResult<Asset> {
+    let all_time = if accrued {
+        None
+    } else {
+        let fees = deps
+            .querier
+            .query::<ProtocolVaultFeesResponse>(&QueryRequest::Wasm(WasmQuery::Smart {
+                contract_addr: vault.clone(),
+                msg: to_binary(&vault_network::vault::QueryMsg::ProtocolFees { all_time: true })?,
+            }))?
+            .fees;
+
+        Some(fees)
+    };
+
+    let mut asset = deps
+        .querier
+        .query::<ProtocolVaultFeesResponse>(&QueryRequest::Wasm(WasmQuery::Smart {
+            contract_addr: vault,
+            msg: to_binary(&vault_network::vault::QueryMsg::ProtocolFees { all_time: false })?,
+        }))?
+        .fees;
+
+    if let Some(all_time_fees) = all_time {
+        asset.amount = all_time_fees.amount.checked_sub(asset.amount)?;
```

Comment:
@kerber0x please, have look at this - I assume that fee_collector collected fees, those sitting on fee_collector, can be obtained by contract.fees(all_time=true) - contract.fees(all_time=false). Is it true?